app/com/linkedin/drelephant/util/Utils.java

```diff
@@ -102,7 +102,7 @@ public static Document loadXMLDoc(String filePath) {
     Map<String, String> options = new HashMap<String, String>();
     String[] tokens = str.trim().split("\\s");
     for (String token : tokens) {
-      if (token.isEmpty()) {
+      if (token.isEmpty() || token.startsWith("-X")) {
```

Comment:
in short these options like `-XX:+UseCompressedOops` are standard JVM params used for tunning JVM performance or reliability (e.g. many Spark (or Hadoop app) Tuning guides refer to these).

if you are interested in details:

>  From `Tuning Spark`: If you have less than 32 GB of RAM [per JVM], set the JVM flag -XX:+UseCompressedOops to make pointers be four bytes instead of eight.

java has a low default for MaxPermSize do like 64mb.

> The permanent space is where the classes, methods, internalized strings, and similar objects used by the VM are stored and never deallocated (hence the name).
> advises you to augment this space if you load many classes (this is typically the case for application servers and some IDE like Eclipse). [and I’d say same applies for code generation]
>    http://stackoverflow.com/questions/32497078/spark-sql-is-throwing-permgen-space-error